app/models/sample.rb

```diff
@@ -2,4 +2,7 @@ class Sample < ApplicationRecord
   belongs_to :project
   has_many :pipeline_outputs, dependent: :destroy
   has_and_belongs_to_many :backgrounds
+  has_many :input_files, dependent: :destroy
+
```

Comment:
1. would be nice to have a helper function to figure out the s3 path of the input file.  like s3_path 
2. all our input data are paired fastq.gz, what do you envision the uploading process?  two input_files entries attached to a sample? or one entry corresponding to a s3 directory?

Reply:
1. for sure, we can add that

2. I decided not to validate the uploads for now, bc I didn't know if we'd have more diverse input times (no paired reads?). I think we should hold off for now until we can get a better sense for the diversity of inputs.

Reply:
2.cont: that said, if our pipeline can only handle paired-end reads then we should probably reject anything that doesn't look like paired end read fastq files

Reply:
we only accept paired reads for now